File: python/baseDatos/deserializador.py
import pickle
import os
import pathlib

def deserializar(aeropuerto):
    #Apertura del archivo donde será leido el flujo de bytes que representen el objeto. En este caso se trata de administracion
    ruta=open(os.path.join(pathlib.Path(__file__).parent.absolute(), "temp\\aeropuerto"),"rb")
    #Indicamos la variable donde se guardará el objeto deserializado. En este caso el valor de la caja de Administración
    aertemp=pickle.load(ruta)
    aeropuerto.setNombre(aertemp.getNombre())
    aeropuerto.setVuelos(aertemp.getVuelos())
    aeropuerto.setPasajeros(aertemp.getPasajeros())
    aeropuerto.setEmpleados(aertemp.getEmpleados())
    aeropuerto.setAviones(aertemp.getAviones())
    aeropuerto.setTransaccionesKeys(aertemp.getTransaccionesKeys())
    aeropuerto.setTransaccionesValues(aertemp.getTransaccionesValues())
    aeropuerto.setDinero(aertemp.getDinero())
    #Se cierra el archivo abierto
    ruta.close()

File: src/basedatos/Deserializador.java
// Autores: Juan Jose Zapata Cadavid
/* 
 * Extraer los valores guardados en la memoria para usarlo luego en la RAM
 */
package basedatos;

import java.io.FileInputStream;
import java.io.FileNotFoundException;
import java.io.IOException;
import java.io.ObjectInputStream;

import gestorAplicacion.gestionVuelos.Aeropuerto;

public class Deserializador {
	/*
	 * Se encarga de extraer a aeropuerto de aeropuerto.txt y asignar todos los
	 * valores al aeropuerto que se pasa por parametro
	 */
	public static <E> void deserializar(Aeropuerto aeropuerto) {
		FileInputStream fin;

		try {
			String ruta = System.getProperty("user.dir") + "\\src\\basedatos\\temp\\aeropuerto.txt";
			fin = new FileInputStream(ruta);
			ObjectInputStream oos = new ObjectInputStream(fin);
			Aeropuerto e = (Aeropuerto) oos.readObject();

			aeropuerto.setAviones(e.getAviones());
			aeropuerto.setDinero(e.getDinero());
			aeropuerto.setEmpleados(e.getEmpleados());
			aeropuerto.setNombre(e.getNombre());
			aeropuerto.setPasajeros(e.getPasajeros());
			// aeropuerto.setSalas(e.getSalas());
			aeropuerto.setVuelos(e.getVuelos());
			aeropuerto.setTransaccionesKeys(e.getTransaccionesKeys());
			aeropuerto.setTransaccionesValues(e.getTransaccionesValues());

			oos.close();
			fin.close();

		} catch (FileNotFoundException e) {
			e.printStackTrace();
		} catch (IOException e) {
			// TODO: handle exception
			e.printStackTrace();
		} catch (ClassNotFoundException e) {
			// TODO: handle exception
			e.printStackTrace();
		}
	}
}
